stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/channel/add/group/select_name/AddGroupChannelSelectNameViewModel.kt

```diff
@@ -0,0 +1,53 @@
+package io.getstream.chat.ui.sample.feature.channel.add.group.select_name
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.livedata.ChatDomain
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class AddGroupChannelSelectNameViewModel : ViewModel() {
+
+    private val currentUserId: String = ChatDomain.instance().currentUser.id
+    private val _state: MutableLiveData<State> = MutableLiveData()
+    val state: LiveData<State> = _state
+
+    fun onEvent(event: Event) {
+        when (event) {
+            is Event.CreateChannel -> createChannel(event.name, event.members)
+        }
+    }
+
+    private fun createChannel(name: String, members: List<User>) {
+        _state.value = State.Loading
+        viewModelScope.launch(Dispatchers.IO) {
+            val result = ChatClient.instance()
+                .createChannel(
+                    channelType = CHANNEL_TYPE_MESSAGING,
+                    members = members.map { it.id } + currentUserId,
+                    extraData = mapOf(EXTRA_DATA_CHANNEL_NAME to name)
+                ).execute()
```

Comment:
We're in a coroutine here, so you can use `await()` instead of `execute()` to run this in a non-blocking way. If you do that, you can also start the coroutine on `DispatcherProvider.Main`, and use `setValue` instead of `postValue` on the `LiveData`.